scripted-tests/run/build-library/src/main/c/testlib.cpp

```diff
@@ -0,0 +1,57 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <assert.h>
+#include <string.h>
+#include "libtest.hpp"
+
+int main() {
+    assert(ScalaNativeInit() == 0);
+
+    sayHello();
+
+    assert(strcmp(native_constant_string(), "ScalaNativeRocks!") == 0);
+
+    assert(native_number() == 42);
+    native_set_number(84);
+    assert(native_number() == 84);
+
+    assert(add_longs(123456789L, 876543210L) == 999999999L);
+
+    struct Foo *p = retStructPtr();
+    assert(p != NULL);
+    assert(p->arg1 == 42);
+    assert(p->arg2 == 2020);
+    assert(p->arg3 == 27);
+    assert(p->arg4 == 14.4556);
+    assert(strcmp(p->arg5, "ScalaNativeRocks!") == 0);
+
+    updateStruct(p);
+    assert(p != NULL);
+    assert(p->arg1 == 42);
+    assert(p->arg2 == 2021);
+    assert(p->arg3 == 27);
+    assert(p->arg4 == 14.4556);
+    assert(strcmp(p->arg5, "ScalaNativeRocks!") == 0);
+    free(p);
+
+    sn_runGC();
+
+    bool exceptionCaught = false;
+    try {
+        fail();
+    } catch (const scalanative::ExceptionWrapper &e) {
+        exceptionCaught = true;
+    }
+    assert(exceptionCaught);
+
+    exceptionCaught = false;
+    try {
+        fail();
+    } catch (const std::exception &e) {
+        exceptionCaught = true;
+    }
+    assert(exceptionCaught);
```

Comment:
@adampauls On Linux, both of these variants seemed to work correctly, if it would still be failing (in the CI) I would try to investigate it more.

Reply:
Hmm, I'm unable to run the tests now, they fail for other reasons (as they do in CI). Should I assume you're fixing that up now, or should I take a look?

Reply:
Don't worry about that, I have it fixed already. Would push the changes tomorrow in the morning